### DISCUSSION TITLE
Two questions

### DISCUSSION THREAD

**Discussion by bloodteller123** [CXwudi/comp5903-easy-cucumber#10] [Q&A] [answered]
Hi Charles,
Thanks for the great works. 

I'd like to ask 
1. Is it officially acceptable to execute a main method to run the test? For example in ```main(...)``` of ```runner class```, could I create instances of ```JFeature``` and then call ```. executeAll()```?
2. if 1) holds, what indications do you have to show the state of different test cases? For instance, If the console prints ```Process finished with exit code 0```, would that mean my results match expected results?
Thank you,
B.T.

**Comment by CXwudi** [comment, marked as answer]
For 1), do you mean executing easy-cucumber's methods in Java `main()` method? Yes you can
For 2), I believe in Java, if a Java program failed with an exception, its exit code would not be 0. I think this is more or less a Java question which is outside the scope of this tool and you can find more detailed answer through google

**Reply by bloodteller123** [reply]
1) Thank you
2) Let me rephrase the question: if ```Process finished with exit code 0``` after calling ```feature.executeAll()```, then all my tests have passed ; Is it true?

**Reply by CXwudi** [reply]
Yes, assuming a test passing iff no exception thrown, which is what JUnit does
